### ISSUE TITLE
Chocolatey reinstalles every chef run

### ISSUE THREAD

**Issue by spuder** [chocolatey/chocolatey-cookbook#105]
Since chocolatey 0.10 has been released, I notice that chocolatey gets installed every time chef runs

```
  * template[c:/chef/cache/install.ps1] action create[2016-09-19T12:02:23-04:00] INFO: Processing template[c:/chef/cache
/install.ps1] action create (chocolatey::default line 29)
 (up to date)
  * powershell_script[Install Chocolatey] action run[2016-09-19T12:02:23-04:00] INFO: Processing powershell_script[Insta
ll Chocolatey] action run (chocolatey::default line 36)

    [execute]
              7-Zip (A) 9.20  Copyright (c) 1999-2010 Igor Pavlov  2010-11-18

              Processing archive: C:\Users\sowen\AppData\Local\Temp\2\chocolatey\chocInstall\chocolatey.zip

              Extracting  _rels\.rels
              Extracting  chocolatey.nuspec
              Extracting  tools\chocolateyInstall.ps1
              Extracting  tools\chocolateysetup.psm1
              Extracting  tools\init.ps1
              Extracting  tools\chocolateyInstall\choco.exe
              Extracting  tools\chocolateyInstall\choco.exe.ignore
              Extracting  tools\chocolateyInstall\LICENSE.txt
              Extracting  tools\chocolateyInstall\helpers\chocolateyInstaller.psm1
              Extracting  tools\chocolateyInstall\helpers\chocolateyProfile.psm1
              Extracting  tools\chocolateyInstall\helpers\chocolateyScriptRunner.ps1
              Extracting  tools\chocolateyInstall\helpers\ChocolateyTabExpansion.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Format-FileSize.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-CheckSumValid.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-ChocolateyUnzip.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-ChocolateyWebFile.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-EnvironmentVariable.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-EnvironmentVariableNames.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-FtpFile.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-OSArchitectureWidth.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-ToolsLocation.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-UACEnabled.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-UninstallRegistryKey.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-VirusCheckValid.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-WebFile.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-WebFileName.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Get-WebHeaders.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-BinFile.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyDesktopLink.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyEnvironmentVariable.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyExplorerMenuItem.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyFileAssociation.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyInstallPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyPath.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyPinnedTaskBarItem.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyPowershellCommand.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyShortcut.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyVsixPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-ChocolateyZipPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Install-Vsix.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Set-EnvironmentVariable.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Set-PowerShellExitCode.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Start-ChocolateyProcessAsAdmin.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Test-ProcessAdminRights.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Uninstall-BinFile.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Uninstall-ChocolateyEnvironmentVariable.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Uninstall-ChocolateyPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\UnInstall-ChocolateyZipPackage.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Update-SessionEnvironment.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Write-ChocolateyFailure.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Write-ChocolateySuccess.ps1
              Extracting  tools\chocolateyInstall\helpers\functions\Write-FileUpdateLog.ps1
              Extracting  tools\chocolateyInstall\redirects\choco.exe
              Extracting  tools\chocolateyInstall\redirects\choco.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\chocolatey.exe
              Extracting  tools\chocolateyInstall\redirects\chocolatey.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cinst.exe
              Extracting  tools\chocolateyInstall\redirects\cinst.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\clist.exe
              Extracting  tools\chocolateyInstall\redirects\clist.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cpack.exe
              Extracting  tools\chocolateyInstall\redirects\cpack.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cpush.exe
              Extracting  tools\chocolateyInstall\redirects\cpush.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cuninst.exe
              Extracting  tools\chocolateyInstall\redirects\cuninst.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cup.exe
              Extracting  tools\chocolateyInstall\redirects\cup.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\cver.exe
              Extracting  tools\chocolateyInstall\redirects\cver.exe.ignore
              Extracting  tools\chocolateyInstall\redirects\RefreshEnv.cmd
              Extracting  tools\chocolateyInstall\tools\7z.dll
              Extracting  tools\chocolateyInstall\tools\7z.exe
              Extracting  tools\chocolateyInstall\tools\7z.exe.ignore
              Extracting  tools\chocolateyInstall\tools\7z.exe.manifest
              Extracting  tools\chocolateyInstall\tools\7zip.license.txt
              Extracting  tools\chocolateyInstall\tools\checksum.exe
              Extracting  tools\chocolateyInstall\tools\checksum.exe.config
              Extracting  tools\chocolateyInstall\tools\checksum.exe.ignore
              Extracting  tools\chocolateyInstall\tools\checksum.license.txt
              Extracting  tools\chocolateyInstall\tools\detector.zip
              Extracting  tools\chocolateyInstall\tools\shimgen.exe
              Extracting  tools\chocolateyInstall\tools\shimgen.exe.ignore
              Extracting  tools\chocolateyInstall\tools\shimgen.license.txt
              Extracting  package\services\metadata\core-properties\b71b0b2f574b4729aa2134477128b378.psmdcp
              Extracting  [Content_Types].xml

              Everything is Ok

              Files: 87
              Size:       11599385
              Compressed: 6894450
              WARNING: It's very likely you will need to close and reopen your shell
                before you can use choco.
              WARNING: You can safely ignore errors related to missing log files when
                upgrading from a version of Chocolatey less than 0.9.9.
                'Batch file could not be found' is also safe to ignore.
                'The system cannot find the file specified' - also safe.
              Downloading Chocolatey from : https://chocolatey.org/api/v2/package/chocolatey

              Mode                LastWriteTime     Length Name

              ----                -------------     ------ ----

              d----         9/19/2016  12:02 PM            chocInstall

              Downloading https://chocolatey.org/api/v2/package/chocolatey to C:\Users\sowen\AppData\Local\Temp\2\chocol
atey\chocInstall\chocolatey.zip
```

**Comment by mwrock** [comment]
The cookbook always calls the chocolatey installer unless the `['chocolatey']['upgrade']` attribute is set to `false`. I'm not sure but maybe the chocolatey installer did a version check before to avoid the actual install and that no longer happens.

**Comment by spuder** [comment]
We got banned today from chocolatey.org because our humble group of 100 servers were doing 14,000 installs per day. 

I think the chocolatey.org maintainers would appreciate chocolatey not trying to reinstall every chef run.

**Comment by ferventcoder** [comment]
Any chance we can get some movement on this? It attempts to install Chocolatey every time.

**Comment by spuder** [comment]
Unfortunately the reason that chocolatey reinstalls every time is because the library helper `chocolatey_installed?` does not return correctly, which results in the `not_if` guard interpreter on this line never executing. Because of the &&, not even setting upgrade = false will prevent the script from running

```ruby
powershell_script 'Install Chocolatey' do
  environment node['chocolatey']['install_vars']
  cwd Chef::Config['file_cache_path']
  code install_ps1
  not_if { chocolatey_installed? && (node['chocolatey']['upgrade'] == false) }
end
```

**Comment by spuder** [comment]
MR #122 gives us the desired behavior where chocolatey doesn't try and update every chef run. 

Could we release 1.1.2 or 1.2.0

**Comment by mwrock** [comment]
Released 1.2.0 last night.
